### PR TITLE
Re-activate vm pipelines

### PR DESCRIPTION
**Pull request by galiacheng**
This pr is to re-activate vm pipelines with the following fix:
- Remove the AZ CLI action, replacing with shell step to avoid az login issue.
- Fix az cli to delete resources in the delete-node template for cluster and dynamic cluster offer
- Fix creating NRMS-Rule-101 issue

**Notes: All the deployment in cluster pipeline will fail as there is an issue in `weblogic-cafe`.**

Steps to set up secrets:
- Install Azure CLI and GitHub CLI locally
- Clone this repo to your local machine
- Login to Azure properly using Azure CLI by running az login
- Login to GitHub using GitHub CLI by runninggh auth login
- Run cd <path to repo>/weblogic-azure/.github/workflows
- Run ./setupForWlsVm.sh
- Type in your credentials and repo info as asked by prompt message
- Got GitHub repo's Setting page: https://github.com/oracle/weblogic-azure/settings
- Check all Secrets are successfully created
- Go to GitHub repo's Action page: https://github.com/oracle/weblogic-azure/actions

You may not have permission to create `AZURE_CREDENTIALS`, I have shared one with Thiru before.

### CODE REVIEW

File: .github/workflows/buildWlsVm4CcArtifact.yml [hunk 1]
```diff
@@ -13,11 +13,12 @@ on:
       ref:
         description: 'Specify Git Ref if needed.'
         required: false
-        default: 'refs/heads/develop'
+        default: 'refs/heads/main'
 env:
   refArmttk: d97aa57d259e2fc8562e11501b1cf902265129d9
-  refJavaee: 6addd99d8bc3f472e040f11c053a37e1ac370229
-  repoName: "arm-oraclelinux-wls-cluster"
+  refJavaee: 13fe6ec487024eb61355d661ab5700ae90cb0a8f
```
Comment (edburns) [inline .github/workflows/buildWlsVm4CcArtifact.yml:19]: Pin to az CLI version 2.30.0.

File: .github/workflows/buildWlsVm4DcArtifact.yml [hunk 2]
```diff
@@ -13,11 +13,12 @@ on:
...
         required: false
-        default: 'refs/heads/develop'
+        default: 'refs/heads/main'
 env:
   refArmttk: d97aa57d259e2fc8562e11501b1cf902265129d9
-  refJavaee: 6addd99d8bc3f472e040f11c053a37e1ac370229
-  repoName: "arm-oraclelinux-wls-dynamic-cluster"
+  refJavaee: 13fe6ec487024eb61355d661ab5700ae90cb0a8f
+  offerName: "arm-oraclelinux-wls-dynamic-cluster"
+  repoName: "weblogic-azure"
```
Comment (edburns) [inline .github/workflows/buildWlsVm4DcArtifact.yml:21]: Pin to az CLI version 2.30.0.

File: .github/workflows/buildWlsVm4SnArtifact.yml [hunk 3]
```diff
@@ -12,11 +12,12 @@ on:
...
         required: false
-        default: 'refs/heads/develop'
+        default: 'refs/heads/main'
 env:
   refArmttk: d97aa57d259e2fc8562e11501b1cf902265129d9
-  refJavaee: 6addd99d8bc3f472e040f11c053a37e1ac370229
-  repoName: "arm-oraclelinux-wls-admin"
+  refJavaee: 13fe6ec487024eb61355d661ab5700ae90cb0a8f
+  offerName: "arm-oraclelinux-wls"
+  repoName: "weblogic-azure"
```
Comment (edburns) [inline .github/workflows/buildWlsVm4SnArtifact.yml:20]: Pin to az CLI version 2.30.0.

File: .github/workflows/setupForWlsVm.sh [hunk 4]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env bash
```
Comment (edburns) [inline .github/workflows/setupForWlsVm.sh:1]: Yes. Thanks for adding this and the teardown.

File: .github/workflows/testWlsVmAdmin.yml [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+  adminConsolePort: 7005
+  adminVMName: adminServerVM
+  adminPassword: ${{ secrets.WLS_PSW }}
+  dbName: wlsdb${{ github.run_id }}${{ github.run_number }}
+  elkURI: ${{ secrets.ELK_URI }}
+  elkUser: ${{ secrets.ELK_USER_NAME }}
+  elkPassword: ${{ secrets.ELK_PSW }}
+  gitToken: ${{ secrets.GIT_TOKEN }}
+  location: eastus
+  nsg: wls-nsg
```
Comment (edburns) [inline .github/workflows/testWlsVmAdmin.yml:29]: Pin to az CLI version 2.30.0.

File: .github/workflows/testWlsVmCluster.yml [hunk 6]
```diff
@@ -2,49 +2,51 @@
 #Released under the Universal Permissive License v1.0 as shown at
 # https://oss.oracle.com/licenses/upl/
 
-name: Build and Test
+name: Test Configured Cluster on VM
```
Comment (edburns) [inline .github/workflows/testWlsVmCluster.yml:5]: Pin to az CLI version 2.30.0.

File: .github/workflows/testWlsVmDynamicCluster.yml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 #Released under the Universal Permissive License v1.0 as shown at
 # https://oss.oracle.com/licenses/upl/
 
-name: Build and Test
+name: Test Dynamic Cluster on VM
```
Comment (edburns) [inline .github/workflows/testWlsVmDynamicCluster.yml:5]: Pin to az CLI version 2.30.0.

**Review by edburns — APPROVED** [review]
Looks good to me. I will advocate for seeing this merged.

File: .github/workflows/setupForWlsVm.sh [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+OTN_USERID=
+# Password for preceding Oracle single sign-on userid.
+OTN_PASSWORD=
+# User Email of GitHub acount to access GitHub repository.
+USER_EMAIL=
+# User name for preceding GitHub account.
+USER_NAME=
+# Personal token for preceding GitHub account.
+GIT_TOKEN=
+WLS_PSW=Secret123456
```
Comment (mriccell) [inline .github/workflows/setupForWlsVm.sh:39]: Is this a secret name or a passwd?

Comment (edburns) [reply .github/workflows/setupForWlsVm.sh:39]: Not really, but I'll fix it to prompt the user for it right now.

Comment (edburns) [reply .github/workflows/setupForWlsVm.sh:39]: Good catch. I am fixing this now.

**Review by mriccell — APPROVED** [review]

**Pull request merged**